### ISSUE TITLE
vim-jsx + es.next.syntax.vim + yajs syntax error for nested JS code in JSX

### ISSUE THREAD

**Issue by rishitank** [othree/yajs.vim#168]
Hi, not sure if this issue belongs in es.next.syntax.vim or yajs, but this issue appears to occur when all 3 plugins are used as you can see below:

![image](https://user-images.githubusercontent.com/3771081/46575238-a766fa80-c9a9-11e8-8387-58bdf7936c39.png)

I use es.next.syntax.vim to add syntax highlighting for decorators, static fields etc.

From what I can tell, it seems that wrapping JSX with brackets (from optgroup above) causes the syntax to be read as if function arguments, hence matching the same colour as function arguments.

**Comment by rishitank** [comment]
I can also confirm that adding vim-jsx does not help.

**Comment by othree** [comment]
Can you provide code snippets for this issue?

I believe this is caused by arrow function. You can see the arrow is not highlighted correctly.

**Comment by rishitank** [comment]
@othree yes it seems the issue is triggered by the arrow symbol and causes the entire block of JSX to be considered as javaScriptFuncArg / jsFuncArgs, therefore anything inside the arrow function that doesn't conform to this causes these syntax errors. I managed to produce a minimal example:

`        const arr = [];

        return (
            <div>
                {
                    arr.map((item, i) => {
                        const foo = item.bar;

                        return (<div></div>);
                    })
                }
            </div>
        );
`

While typing this, as soon as I entered the arrow symbol, everything inside the brackets of the top level return statement seems to be parsed as if they were function arguments.

**Comment by rishitank** [comment]
Update: I've found that if there is only one argument in the arrow function of a nested JSX expression then the syntax highlighting and errors do not occur, but if wrapping the function arguments in brackets e.g. if specifying multiple arguments such as `(item, i) =>`, then the above issue occurs, whereas `item =>` is valid.
